docs/source/cost_functions.rst

```diff
@@ -0,0 +1,80 @@
+.. Copyright 2018 The Cartographer Authors
+
+.. Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+..      http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+==============
+Cost functions
+==============
+
+Relative Transform Error 2D
```

Comment:
Naming should be consistent with the code (mention SpaCostFunction2D or the like)

Reply:
Sure, I planned to rename the `SpaCostFunction` to `RelativeTransformError`. I think it makes more sense.

Reply:
@wally-the-cartographer merge